server/server.js

```diff
@@ -57,6 +57,8 @@ let needSetup = false;
 
     console.log("Adding route")
     app.use('/', express.static("dist"));
+    app.use(apiMetrics())
```

Comment:
Is it safe that expose the api like this?

Reply:
@louislam it's bad wording - this is a read-only endpoint which exposes metrics about how the api is performing, but it's not something I can rename because it's build into the library that provides this function.

Here's some sample output from `http://localhost:3001/metrics` on my dev setup:

```
# HELP monitor_response_time Monitor Response Time (ms)
# TYPE monitor_response_time gauge
monitor_response_time{monitor_name="Google",monitor_type="http",monitor_url="https://www.google.com",monitor_hostname="null",monitor_port="null"} 202
monitor_response_time{monitor_name="BBC",monitor_type="http",monitor_url="https://www.bbc.co.uk",monitor_hostname="null",monitor_port="null"} 259

# HELP monitor_status Monitor Status (1 = UP, 0= DOWN)
# TYPE monitor_status gauge
monitor_status{monitor_name="Google",monitor_type="http",monitor_url="https://www.google.com",monitor_hostname="null",monitor_port="null"} 1
monitor_status{monitor_name="BBC",monitor_type="http",monitor_url="https://www.bbc.co.uk",monitor_hostname="null",monitor_port="null"} 1

# HELP process_cpu_user_seconds_total Total user CPU time spent in seconds.
# TYPE process_cpu_user_seconds_total counter
process_cpu_user_seconds_total 0.23349

# HELP process_cpu_system_seconds_total Total system CPU time spent in seconds.
# TYPE process_cpu_system_seconds_total counter
process_cpu_system_seconds_total 0.043258

# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 0.276748

# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1626951734

# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 57819136

# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 927924224

# HELP process_heap_bytes Process heap size in bytes.
# TYPE process_heap_bytes gauge
process_heap_bytes 99782656

# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 25
```

Reply:
As for now, Uptime Kuma is like UptimeRobot, users need to login to view any data.

If I read the source code correctly, anyone go to "/metrics" can read the data?

Reply:
hmmmm, I'd not thought about that, ok, this is where my vague attempts at Javascript fail me - any idea how I might protect this?

Prometheus can easily send username/password for basicAuth etc, but I don't know enough about the express framework to see how we might protect this...

Reply:
You can leave this part to me.